jsx-runtime.js

```diff
@@ -0,0 +1,43 @@
+const { createElement, Fragment } = require('./index');
+
+// This is a helper function to ensure that children is always an array
+/**
+ * @type {(
+ *   children: import('./index.d.ts').Children
+ * ) => import('./index.d.ts').Children[]}
+ */
+function safeChildrenArray(children) {
+  if (!children) {
+    return [];
+  }
+  if (Array.isArray(children)) {
+    return children;
+  } else {
+    return [children];
+  }
+}
+
+// The jsx function is the main function that will be used to create elements
+// when the jsx-runtime is used
+/** @type {import('./jsx-runtime').jsx} */
+function jsx(type, { children, ...props }) {
+  if (typeof type === 'string') {
+    // This is needed because the content of the element (aka children)
+    // must be provided as sequential arguments to createElement
+    // so in order to spread it safely we need to ensure that it is an array
+    const childrenArray = safeChildrenArray(children);
+    return createElement(type, props, ...childrenArray);
```

Comment:
`createElement` childrens can be a infinite depth array, or a single item. Because of the below implementation, `contentsToString` flatterns on the go the provided array and simply work.

https://github.com/kitajs/html/blob/d556fcddfff0337b1f0ec0b8b2a81f1069d45155/index.js#L395-L399

Reply:
I'm not sure that I follow you on this one. The issue is that the createElement function expects children to be passed as sequential parameters after the type of tag and the props. Maybe we should change the signature of the createElement function to support both cases. But again, I don't think I have understood your point here.